squash/squash/views.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+from django.conf import settings
+from django.http import Http404
+from django.shortcuts import render
+from django.template import Template
+from django.utils._os import safe_join
+
+
+def get_page_or_404(file_name):
+    """Return page content as a Django template or raise 404 error.
+    :param name: name of the page to display
+    """
+    try:
+        file_path = safe_join(settings.SITE_PAGES_DIRECTORY, file_name)
+    except ValueError:
+        raise Http404('Page Not Found')
+    else:
+        if not os.path.exists(file_path):
+            raise Http404('Page Not Found')
+    with open(file_path, 'r') as f:
+        page = Template(f.read())
+
+    return page
```

Comment:
I'd use the template API to do this now. It's relatively new but definitely a better solution after it's implemented. Just ask if you need some help working out the details.

https://docs.djangoproject.com/en/1.8/topics/templates/#usage